components/brave_wallet/browser/cardano/cardano_create_transaction_task.h

```diff
@@ -0,0 +1,80 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_CARDANO_CARDANO_CREATE_TRANSACTION_TASK_H_
+#define BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_CARDANO_CARDANO_CREATE_TRANSACTION_TASK_H_
+
+#include <map>
+#include <string>
+#include <utility>
+
+#include "base/memory/weak_ptr.h"
+#include "base/types/expected.h"
+#include "brave/components/brave_wallet/browser/cardano/cardano_rpc_schema.h"
+#include "brave/components/brave_wallet/browser/cardano/cardano_transaction.h"
+#include "brave/components/brave_wallet/common/brave_wallet.mojom.h"
+
+namespace brave_wallet {
+class CardanoWalletService;
+class CardanoTransaction;
+
+// This class implements `CardanoWalletService::CreateTransaction` logic of
+// creating a Cardano transaction based on wallet's account, destination address
+// and amount of native coins to send. Fetches latest block and epoch
+// parameters, utxos associated with account . Searches for best utxo set to
+// minimize fee. Responds with to-be-signed transaction to `callback_`.
+class CreateCardanoTransactionTask {
+ public:
+  using UtxoMap = std::map<CardanoAddress, cardano_rpc::UnspentOutputs>;
+  using Callback =
+      base::OnceCallback<void(base::expected<CardanoTransaction, std::string>)>;
+
+  CreateCardanoTransactionTask(CardanoWalletService& cardano_wallet_service,
+                               const mojom::AccountIdPtr& account_id,
+                               const CardanoAddress& address_to,
+                               uint64_t amount,
+                               bool sending_max_amount);
+
+  ~CreateCardanoTransactionTask();
+
+  void set_callback(Callback callback) { callback_ = std::move(callback); }
+
+  void ScheduleWorkOnTask();
```

Comment:
why can't we do `ScheduleWorkOnTask(Callback callback)`? same question applied to `GetCardanoUtxosTask`

Reply:
Added Start(callback) method